### DISCUSSION TITLE
Setting manual job_id

### DISCUSSION THREAD

**Discussion by LSnyd** [NVIDIA/NVFlare#755] [Q&A] [answered]
Hi all, 
I've updated my project to the latest NVFlare version and one of the changes includes using UUIDs as job ids instead of using a manually given run_number as in the previous version. 

The UUIDs are a bit unclear to me, as it is much harder to organize my runs like that. 
Before I could for instance call all cross-validation runs with Fedavg 200-205 and all with FedOpt 300-305. 
So it was a bit more clear where I can find certain experiments. With the random UUIDs, I would have to check the config files of all jobs to find a certain configuration.

Is there still a way to add a manual job_id? I tried to find the position in NVFlare, but couldn't find where the job id is ultimately set.

**Comment by YuanTingHsieh** [comment]
Hi @LSnyd thanks for bringing this up.
When you say "latest" do you mean the latest version on Pypi?
It will be more clear to others if you can provide the version number.

For the released 2.1 version, I think we DO NOT have a way to set manual job id.
 @yhwen please correct me if I am wrong.
 
 @yanchengnv @chesterxgchen @holgerroth This feature is worth discussing if the users want to add manual job id / name to identify their job.

**Comment by holgerroth** [comment, marked as answer]
The user can already define a manual job_name inside meta.json, see [here](https://github.com/NVIDIA/NVFlare/blob/4fa2343819bba12e219f4f2587d293ed8492bb1a/examples/cifar10/job_configs/cifar10_fedavg/meta.json#L2). 
The job_id is a UUID and should be defined by the system (nvflare) to allow unique identification.
